### PR TITLE
Feature/overdue debts

### PR DESCRIPTION
**Pull request by NinoRy**
#

### CODE REVIEW

File: modules/OverdueDebts/Entities/DebtImport.php [hunk 1]
```diff
@@ -96,99 +109,173 @@ public function run()
...
+        $indicator = null;
+        if ($line[self::INDICATOR] > 0 && $line[self::INDICATOR] <= 3) {
+            // Add dunning charge to each debt separately - kept here in case it's needed again in future
+            // $fee = $this->conf->{'dunning_charge'.$indicator};
+
             $indicator = $line[self::INDICATOR];
-            $fee = $this->conf->{'dunning_charge'.$indicator};
-        } elseif ($line[self::INDICATOR] > 4) {
-            $indicator = 4;
+        } elseif ($line[self::INDICATOR] > 3) {
```
Comment (cschra) [inline modules/OverdueDebts/Entities/DebtImport.php]: simple else enough?

File: modules/OverdueDebts/Entities/DebtImport.php [hunk 2]
```diff
@@ -96,99 +111,185 @@ public function run()
...
             // 'number' => $line[self::INVOICE_NR],
             'amount' => str_replace(',', '.', $line[self::AMOUNT]),
-            'missing_amount' => str_replace(',', '.', $line[self::MISSING_AMOUNT]) + $fee,
+            'missing_amount' => str_replace(',', '.', $line[self::MISSING_AMOUNT]),
             'date' => date('Y-m-d', strtotime($line[self::DATE])),
             'dunning_date' => $line[self::DUN_DATE] ? date('Y-m-d', strtotime($line[self::DUN_DATE])) : null,
             'description' => $line[self::DESC],
             'indicator' => $indicator,
-            'total_fee' => $fee,
+            // 'total_fee' => $fee,
```
Comment (cschra) [inline modules/OverdueDebts/Entities/DebtImport.php]: what happened to fee/total fee?

File: modules/OverdueDebts/Resources/lang/en/messages.php [hunk 3]
```diff
@@ -6,7 +6,8 @@
     'clearParentId' => 'Cleared relation to debt as it doesnt make sense as both amounts are positive/negative.',
     'csvImportActive' => 'The import of overdue debts is currently running. Please wait until this process has finished.',
     'import' => [
-        'contractsBlocked' => 'The internet access of the modems of the following contracts was blocked during import: :numbers',
+        'block' => 'Block internet access and add Dunning charges...',
+        'contractsBlocked' => 'The internet access of the modems of the following :count contracts was blocked during import: :numbers',
```
Comment (cschra) [inline modules/OverdueDebts/Resources/lang/en/messages.php]:
```suggestion
        'contractsBlocked' => 'The internet access of the modems of the following :count contracts were blocked during import: :numbers',
```

File: modules/OverdueDebts/Entities/DefaultTransactionParser.php [hunk 4]
```diff
@@ -384,18 +385,18 @@ private function getConf()
      *
      * @return array
      */
-    private function searchNumbers()
+    public function searchNumbers($string)
```
Comment (cschra) [inline modules/OverdueDebts/Entities/DefaultTransactionParser.php]: string is a very arbitrary variable name... :see_no_evil: :speak_no_evil:  :hear_no_evil:

File: modules/OverdueDebts/Entities/DebtImport.php (hunk 1 above)
Comment (NinoRy) [reply modules/OverdueDebts/Entities/DebtImport.php]: no

File: modules/OverdueDebts/Entities/DebtImport.php (hunk 2 above)
Comment (NinoRy) [reply modules/OverdueDebts/Entities/DebtImport.php]: is set later

File: modules/OverdueDebts/Resources/lang/en/messages.php (hunk 3 above)
Comment (NinoRy) [reply modules/OverdueDebts/Resources/lang/en/messages.php]: is wrong imo

File: modules/OverdueDebts/Entities/DefaultTransactionParser.php (hunk 4 above)
Comment (NinoRy) [reply modules/OverdueDebts/Entities/DefaultTransactionParser.php]: imo there is no better one here

File: modules/OverdueDebts/Resources/lang/en/messages.php (hunk 3 above)
Comment (cschra) [reply modules/OverdueDebts/Resources/lang/en/messages.php]: you are right

**Pull request merged**